pennylane/labs/lie/structure_constants.py

```diff
@@ -56,17 +56,15 @@ def structure_constants_dense(g: TensorLike) -> TensorLike:
     dimg, chi, _ = g.shape
 
     # compute all commutators
-    m0m1 = np.einsum("aij,bjk->abik", g, g)
-    m0m1 = np.reshape(m0m1, (-1, chi, chi))
-
-    m1m0 = np.einsum("aij,bki->abkj", g, g)
-    m1m0 = np.reshape(m1m0, (-1, chi, chi))
+    prod = np.einsum("aij,bjk->abik", g, g)
+    m0m1 = np.reshape(prod, (-1, chi, chi))
+    m1m0 = np.reshape(prod.transpose((1, 0, 2, 3)), (-1, chi, chi))
     all_coms = m0m1 - m1m0
 
     # project commutators on basis of g
     # vectorized computation to obtain coefficients in decomposition:
     # v = ∑ (tr(v @ e_j) / ||e_j||^2) * e_j
-    adj = np.tensordot(g, all_coms, axes=[[1, 2], [-1, -2]])
-    adj = -np.reshape(adj, (dimg, dimg, dimg)).imag
-
+    gram_inv = np.linalg.pinv(np.tensordot(g, g, axes=[[1, 2], [2, 1]])).real
```

Comment:
the pinv is coming from the normalization?

Reply:
Yes, in order to compute what the comment says, we need to divide by the norms. I suppose here I'm overdoing it a bit, but we get validity for non-orthonormal matrices for free :)

Reply:
(So we can also process inputs that are not directly coming from `lie_closure_dense`)

Reply:
Not sure that's worth the cost of doing the svd in pinv though 🤔 we can keep for now but remember that it may become a blocker for larger sizes

Reply:
Fair enough. We can add an option to turn this off, i.e. a user can assert that the inputs are orthonormal?

Reply:
Prototyped it, let me know what you think! :)

Reply:
did you push it already? I am still seeing the pinv by default

Reply:
Sorry... Pushed now.